File: frontends/beta-frontend/shared/models/Match.ts
export interface Match {
  id: string;
  home: MatchTeam;
  away: MatchTeam;
  dateTime: string;
  status: MatchStatus;
}

export enum MatchStatusEnum {
  PENDING = 'PENDING',
  ACCEPTED = 'ACCEPTED',
  REJECTED = 'REJECTED',
  SCHEDULED = 'SCHEDULED',
  PLAYED = 'PLAYED',
}
export type MatchStatus =
  | 'PENDING'
  | 'ACCEPTED'
  | 'REJECTED'
  | 'SCHEDULED'
  | 'PLAYED';

export interface MatchTeam {
  id: string;
  modifiers: Modifiers;
}

export interface Modifiers {
  tactic: string;
  horizontalPressure: string;
  verticalPressure: string;
}

File: services/main-service/match-domain/src/main/java/com/kjeldsen/match/domain/entities/Match.java
package com.kjeldsen.match.domain.entities;

import java.time.LocalDateTime;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;
import org.springframework.data.mongodb.core.mapping.Document;

@Builder
@Data
@NoArgsConstructor
@AllArgsConstructor
@ToString
@Document(collection = "Match")
public class Match {

    public enum Status {
        PENDING,
        ACCEPTED,
        REJECTED,
        SCHEDULED,
        PLAYED
    }

    String id;
    String leagueId;
    Team home;
    Team away;
    LocalDateTime dateTime;
    Status status;
    MatchReport matchReport;
}
